testrunner/execute.go

```diff
@@ -213,9 +211,48 @@ func buildTests(lines bytes.Buffer, input_dir string) ([]testResult, error) {
 		return nil, errors.New(pkgLevelMsg)
 	}
 
+	results = addNonExecutedTests(rootLevelTests, results)
+
 	return results, nil
 }
 
+func addNonExecutedTests(rootLevelTests []rootLevelTest, results []testResult) []testResult {
+	insertResultAfterIdx := -1
+	for parentIdx, parentTest := range rootLevelTests {
+		parentFound := false
+		for resultIdx := range results {
+			parentName, _ := splitTestName(results[resultIdx].Name)
+			if rootLevelTests[parentIdx].name == parentName {
+				insertResultAfterIdx = resultIdx
+				parentFound = true
+			}
+		}
+
+		if parentFound {
+			continue
+		}
+
+		newResult := testResult{
+			Name:     parentTest.name,
+			Status:   statErr,
+			TestCode: parentTest.code,
+			Message:  "This test was not executed.",
```

Comment:
I suggest we start with this simple short message for non-executed tests. If students are confused, we can change to something more elaborate later on.